backend/src/security/oauth2.py

```diff
@@ -0,0 +1,84 @@
+"""
+Function overrides necessary for the Swagger documents to use client id and client secret
+instead of a username and password flow.
+"""
+
+from typing import Optional
+
+from fastapi.exceptions import HTTPException
+from fastapi.param_functions import Form
+from fastapi.openapi.models import OAuthFlows as OAuthFlowsModel
+from fastapi.security import HTTPBasic
+from fastapi.security.oauth2 import OAuth2
+from fastapi.security.utils import get_authorization_scheme_param
+
+from starlette.requests import Request
+from starlette.status import HTTP_401_UNAUTHORIZED
+
+from pydantic import BaseModel
+
+# Disabling linting rules as it is FastAPI code just being overridden.
+
+
+class HTTPClientCredentials(BaseModel):
+    """ Defines credentials to use client_id and client_secret over username and password """
+    client_id: str
+    client_secret: str
+
+
+class HTTPBasicClientCredentials(HTTPBasic):  # pylint: disable=too-few-public-methods
+    """ Handles network request for transmitting credentials via Swagger authorize """
+
+    async def __call__(  # type:ignore
+        self, request: Request
+    ) -> Optional[HTTPClientCredentials]:
+        basic = await super().__call__(request)
+        if not basic:
+            return None
+        return HTTPClientCredentials(client_id=basic.username, client_secret=basic.password)
+
+
+class OAuth2ClientCredentials(OAuth2):  # pylint: disable=too-few-public-methods
+    """ Updates the flow from username and password to client credentials for Swagger """
+
+    def __init__(
+        self,
+        tokenUrl: str,
+        scheme_name: str = "oAuth2ClientCredentials",
+        scopes: dict = None,
+        auto_error: bool = True,
+    ):
+        if not scopes:
+            scopes = {}
+        flows = OAuthFlowsModel(clientCredentials={"tokenUrl": tokenUrl, "scopes": scopes})
+        super().__init__(flows=flows, scheme_name=scheme_name, auto_error=auto_error)
+
+    async def __call__(self, request: Request) -> Optional[str]:
+        authorization: str = request.headers.get("Authorization")
+        scheme, param = get_authorization_scheme_param(authorization)
+        if not authorization or scheme.lower() != "bearer":
+            if self.auto_error:  # pylint: disable=no-else-raise
```

Comment:
[https://github.com/tiangolo/fastapi/blob/c59539913dbed20b980f512fcf7d29ce55fa0304/fastapi/security/oauth2.py#L206](https://github.com/tiangolo/fastapi/blob/c59539913dbed20b980f512fcf7d29ce55fa0304/fastapi/security/oauth2.py#L206)

Here's the analog line in the fastapi source code. 

The modified src/security/OAuth2.py file could be fixed to eliminate the linting errors. Or another option, it can be omitted altogether from linting.

Reply:
Lets go ahead and update the code to pass linting.  If we experience some long term adverse affect due to not matching closer to the original source, we can revert the linting ignore rule back on and document what happened in that case.